.github/workflows/pull_push_request.yml

```diff
@@ -0,0 +1,43 @@
+# This is a CI workflow file created to execute on push and pull requests
+
+name: pull_push_workflow
+on: [pull_request, push]
+jobs:
+
+  kotlin-compiler:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Update packages
+        run: sudo apt update
+      - name: Installing JDK
+        run: sudo apt install default-jdk
+      - name: Downloading kotlin compiler
+        run: sudo snap install --classic kotlin
+      - name: Message
+        run: echo "kotlin complier installation complete...."
+
+  make-install:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install make
+        run: sudo apt install make -y
+      - name: Checking make version
+        run: make --version
+      - name: Checking the make directory
+        run: ls /usr/bin/make
+      - name: Message
+        run: echo "makefile installation complete...."
+
+  run-make:
+    runs-on: ubuntu-latest
+    needs: make-install
+    defaults:
+      run:
+        working-directory: tests
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v3
+      - name: Executing make
+        run: make run
```

Comment:
just run command  `make` instead of `make run` since this command is what will be written in the README once documentation is done

Reply:
Done